---
title: "DE_DMR"
output: html_document
date: "`r Sys.Date()`"
editor_options: 
  chunk_output_type: inline
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(eval = FALSE)
```

```{r, message=F, warning=F}
library(tidyverse)
library(ggnewscale)
library(ggpp)
library(RVenn)
library(XGR)
library(showtext)
```

```{r}
dir.create('Barplots')
```

# DMR

## Read gene tables

```{r}
dmr_tables_CpG = readRDS('RDS_files/dmr_tables_CpG.rds')
```


```{r}
n_sigs_dmr <- sapply(dmr_tables_CpG$tables_all_overlap, function(x){ x$gene %>% unique() %>% length() })
# n_sigs_hypo <- sapply(tables_hypo_dmr, nrow)
# n_sigs_hyper<- sapply(tables_hyper_dmr, nrow)

dfm = data.frame(Comparison = titles_dmr, n_genes = n_sigs_dmr)
# dfm_hyper = data.frame(Comparison = titles_dmr, Hypermethylated = n_sigs_hyper)
# dfb_hypo = data.frame(Comparison = titles_dmr, Hypomethylated = n_sigs_hypo)

dfm$group[dfm$Comparison %in% 
            c('AtF3_vs_AtK3', 'AtF5_vs_AtK5', 'AtF5_vs_AtF3', 'AtK5_vs_AtK3')] = 'Сравнения Atalante'
            # c('AtF3_vs_AtK3', 'AtF5_vs_AtK5', 'AtF5_vs_AtF3', 'AtK5_vs_AtK3')] = 'Comparisons of Atalante'
dfm$group[dfm$Comparison %in% 
            c('LMF3_vs_LMK3', 'LMF5_vs_LMK5', 'LMF5_vs_LMF3', 'LMK5_vs_LMK3')] = 'Сравнения LM98'
            # c('LMF3_vs_LMK3', 'LMF5_vs_LMK5', 'LMF5_vs_LMF3', 'LMK5_vs_LMK3')] = 'Comparisons of LM98'
dfm$group[dfm$Comparison %in% 
            c('AtF3_vs_LMF3', 'AtF5_vs_LMF5', 'AtK5_vs_LMK5', 'AtK3_vs_LMK3')] = 'Сравнения между сортами'
            # c('AtF3_vs_LMF3', 'AtF5_vs_LMF5', 'AtK5_vs_LMK5', 'AtK3_vs_LMK3')] = 'Comparisons between varieties'


dfm$group <- as.factor(dfm$group)
dfm$Comparison <- gsub('_vs_', ' vs. ', dfm$Comparison)

dfm =  dfm %>% arrange(group, Comparison)

# dfm2 <- cbind(dfm_hyper, Hypomethylated=dfb_hypo$Hypomethylated)

# dfbm2_long = pivot_longer(data=dfm2, values_to = 'Number_of_rows',
#              cols=c('Hypermethylated', 'Hypomethylated'), 
#              names_to = 'Methylation') %>% 
#   mutate(Methylation = factor(Methylation, levels = c('Hypermethylated', 'Hypomethylated'))) %>%
#   mutate(Comparison = factor(Comparison)) %>% arrange(desc(Methylation))
```

## Barplot DMR

```{r}
# pdf('Barplots/DMR_gene_number_bars_LU.pdf', width = 7, height = 6.5)
png('Barplots/DMR_gene_number_bars_LU.png', width = 2100, height = 1950, units = 'px')

showtext_auto()

ggplot(dfm, aes(x = Comparison, y = n_genes, fill = group)) + geom_bar(stat = 'identity') +
  geom_text(aes(label = n_genes, y = n_genes + 150),position = position_stack(vjust = 1), size = 18) +
  theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 45),
                          axis.text.y = element_blank(),
                            # element_text(angle = 90, vjust = 1, size = 40),
                          legend.text = element_text(size = 60), axis.title = element_text(size = 60),
                          legend.title = element_text(size = 60), legend.position = c(0.75, 0.9)) +
  scale_x_discrete(limits=dfm$Comparison) +
  coord_cartesian(ylim = c(0, 6400)) + 
  scale_y_continuous(expand = c(0,2)) +
  guides(fill = guide_legend(override.aes = list(size = 30) ) ) +
  xlab("\nСравнение") + ylab("Число генов, ассоциированных с ДМР") +  labs(fill=NULL)
  # xlab("Contrast") + ylab("Number of genes associated with DMR") +  labs(fill=NULL)

dev.off()
```

# DE

## FO

```{r}
v_sigs_fo = readRDS('RDS_files/v_sigs_fo.rds')
```

```{r}
n_sigs_fo <- sapply(v_sigs_fo, nrow)

dfb_fo = data.frame(Comparison = rownames(dfb_fo), n_genes = n_sigs_fo) 

# dfb$group = 0

dfb_fo$group[dfb_fo$Comparison %in% 
            c('AtF3_vs_FO', 'AtF5_vs_FO')] = 'Сравнения Atalante'
            # c('AtF3_vs_FO', 'AtF5_vs_FO')] = 'Atalante comparisons'
dfb_fo$group[dfb_fo$Comparison %in% 
            c('LMF3_vs_FO', 'LMF5_vs_FO')] = 'Сравнения LM98'
            # c('LMF3_vs_FO', 'LMF5_vs_FO')] = 'LM98 comparisons'

dfb_fo$Comparison <- gsub('_vs_', ' vs. ', dfb_fo$Comparison)

dfb_fo =  dfb_fo %>% arrange(Comparison)
```


### Barplot DE FO

```{r}
font_add(family = where[1, ]$family, regular = where[1, ]$file)
```

```{r}

# pdf('Barplots/DE_gene_number_bars_FO.pdf', width = 4, height = 4, encoding = 'default', family = 'Times')
png('Barplots/DE_gene_number_bars_FO.png', width = 1200, height = 1200)

showtext_auto()

ggplot(dfb_fo, aes(x = Comparison, y = n_genes, fill= group)) + geom_bar(stat = 'identity') +
  geom_text(aes(label = n_genes, y = n_genes + 100), position = position_stack(vjust = 1), size = 18) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 0, size = 40), 
        axis.text.y = element_blank(),
          # element_text(angle = 90, vjust = 1, size = 30),
        legend.text = element_text(size = 40), 
        axis.title.x = element_text(size = 60), axis.title.y = element_text(size = 40),
        legend.title = element_text(size = 60),
        legend.position = c(0.77, 0.9)) + 
  # scale_x_discrete(limits=dfb$Comparison) +
  coord_cartesian(ylim = c(0, 2000)) + 
  scale_y_continuous(expand = c(0,2)) +
  guides(fill = guide_legend(override.aes = list(size = 20) ) ) +
  xlab("\nСравнение") + ylab("Число дифф. экспрессирующихся генов") + labs(fill=NULL)
  # xlab("Contrast") + ylab("Number of DE genes") + labs(fill=NULL)

dev.off()

```

## LU

```{r}
v_sigs = readRDS('RDS_files/v_sigs.rds')
```

```{r}
n_sigs <- sapply(v_sigs, nrow)

dfb = data.frame(Comparison = titles, n_genes = n_sigs) 

dfb$group[dfb$Comparison %in% 
            # c('AtF3_vs_AtK3', 'AtF5_vs_AtK5', 'AtF5_vs_AtF3', 'AtK5_vs_AtK3')] = 'Сравнения Atalante'
            c('AtF3_vs_AtK3', 'AtF5_vs_AtK5', 'AtF5_vs_AtF3', 'AtK5_vs_AtK3')] = 'Comparisons of Atalante'
dfb$group[dfb$Comparison %in% 
            # c('LMF3_vs_LMK3', 'LMF5_vs_LMK5', 'LMF5_vs_LMF3', 'LMK5_vs_LMK3')] = 'Сравнения LM98'
            c('LMF3_vs_LMK3', 'LMF5_vs_LMK5', 'LMF5_vs_LMF3', 'LMK5_vs_LMK3')] = 'Comparisons of LM98'
dfb$group[dfb$Comparison %in% 
            # c('AtF3_vs_LMF3', 'AtF5_vs_LMF5', 'AtK5_vs_LMK5', 'AtK3_vs_LMK3')] = 'Сравнения между сортами'
            c('AtF3_vs_LMF3', 'AtF5_vs_LMF5', 'AtK5_vs_LMK5', 'AtK3_vs_LMK3')] = 'Comparisons between varieties'

dfb$group <- as.factor(dfb$group)
dfb$Comparison <- gsub('_vs_', ' vs. ', dfb$Comparison)

dfb =  dfb %>% arrange(group, Comparison)
```

```{r}
font_add(family = where[1, ]$family, regular = where[1, ]$file)
```

### Barplot DE LU

```{r}

# pdf('Barplots/DE_gene_number_bars_LU.pdf', width = 7, height = 6.5, encoding = 'default', family = 'Times')
png('Barplots/DE_gene_number_bars_LU.png', width = 1400, height = 1300)

showtext_auto()

n_sigs <- sapply(v_sigs[c('AtF3_vs_AtK3', 'AtF5_vs_AtK5', 'AtF5_vs_AtF3', 'AtK5_vs_AtK3')], nrow)

dfb = data.frame(Comparison = c('AtF3_vs_AtK3', 'AtF5_vs_AtK5', 'AtF5_vs_AtF3', 'AtK5_vs_AtK3'), n_genes = n_sigs)
dfb$Comparison <- gsub('_vs_', ' vs. ', dfb$Comparison)

ggplot(dfb, aes(x = Comparison, y = n_genes)) + geom_bar(stat = 'identity') +
  geom_text(aes(label = n_genes, y = n_genes + 100), position = position_stack(vjust = 1), size = 18) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 0, vjust = 0.5, size = 40), 
        axis.text.y = element_blank(),
          # element_text(angle = 90, vjust = 1, size = 40),
        legend.text = element_text(size = 60), 
        axis.title = element_text(size = 40),
                          legend.title = element_text(size = 60),
                          legend.position = c(0.75, 0.8)) + 
  scale_x_discrete(limits=dfb$Comparison) +
  scale_y_continuous(expand = c(0,2)) +  coord_cartesian(ylim = c(0, 4000)) + 
  guides(fill = guide_legend(override.aes = list(size = 30) ) ) +
  
  xlab("\nСравнение") + ylab("Число дифф. экспрессирующихся генов") + labs(fill=NULL)
  # xlab("Contrast") + ylab("Number of DE genes") + labs(fill=NULL)

dev.off()

```

### up and down (not used)

```{r, eval=F}
n_sigs_down <- sapply(v_sigs_down, nrow)
n_sigs_up <- sapply(v_sigs_up, nrow)

dfb_up = data.frame(Comparison = titles, Upregulated = n_sigs_up)
dfb_down = data.frame(Comparison = titles, Downregulated = n_sigs_down)

dfb <- cbind(dfb_up, Downregulated=dfb_down$Downregulated)


dfb_long = pivot_longer(data=dfb, values_to = 'Number_of_rows',
             cols=c('Upregulated', 'Downregulated'), 
             names_to = 'Regulation') %>% 
  mutate(Regulation = factor(Regulation, levels = c('Upregulated', 'Downregulated'))) %>% 
  mutate(Comparison = factor(Comparison)) %>% arrange(desc(Regulation))
```

```{r eval=F}
dfb_long$Comparison <- gsub('_vs_', ' vs. ', dfb_long$Comparison)
```

```{r eval=F}

pdf('Barplots/DMR_gene_number_bars_LU_up_down.pdf', width = 8, height = 6.5)
ggplot(dfb_long, aes(x = Comparison, y = Number_of_rows, fill = Regulation)) + geom_bar(stat = 'identity') +
  geom_text(aes(label = Number_of_rows, y = Number_of_rows), position = position_stack(vjust = 0.5), size = 3) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 9),
                          legend.text = element_text(size = 15), axis.title = element_text(size = 18),
                          legend.title = element_text(size = 16), title = element_text(size = 17)) + 
  scale_x_discrete(limits=sort(dfb_long$Comparison)) + ylim(0, 4000) + scale_y_continuous(expand = c(0,0)) +
  xlab("Comparison") + ylab("Number of genes") + labs(fill='Dif. expression', title = 'Differential expression of flax genes')

dev.off()

```

## DE and DMR

### Simple barplots

```{r fig.height=5, fig.width=7}
png('Barplots/Big_barplot.png', width = 3000, height = 2400, units = 'px')

showtext_auto()

ggplot() + geom_bar(data=dfb, mapping=aes(x = Comparison, y = n_genes, fill = '0'),
                                          # fill = group), 
                    stat = 'identity', 
                    # linetype = 'dashed', size = 2, col = 'black',
                    position = position_stacknudge(x=-0.22), width = 0.4) +
  
  geom_text(data=dfb, mapping=aes(label = n_genes, 
                                       y = n_genes + 100, 
                                       x = Comparison), 
            # fontface='bold',
            position = position_stacknudge(x=-0.22, vjust = 1, reverse = F), size = 18) +
  # scale_fill_brewer(palette = 'Set1') +
  # new_scale_fill() +
  geom_bar(data=dfm, mapping=aes(x = Comparison, y = n_genes, fill = '1'),
                                 # fill = group),
           # size = 2, col = 'black', 
           stat = 'identity',
           position = position_stacknudge(x=0.22), width = 0.4) +
  
  geom_text(data=dfm, mapping=aes(label = n_genes, 
                                       y = n_genes + 100, 
                                       x = Comparison), 
            # fontface='bold',
            position = position_stacknudge(x=0.22, vjust = 1), size = 18) +
  # scale_fill_brewer(palette = 'Set1') +
  theme_classic() + theme(axis.text.x = element_text(angle = 60, vjust = 0.5, size = 46),
                          axis.text.y = element_text(size = 46, angle = 90),
                        legend.text = element_text(size = 80), axis.title = element_text(size = 70, face = 'bold'),
                        legend.title = element_text(size = 80), 
                        # title = element_text(size = 60),
                        legend.position = c(0.85, 0.4)) +
  
  scale_fill_manual(labels = c('Dif.expressed ', 
                               'Dif.methylated'), 
                    values = c('darkorange', 'lightblue'))  +
  
  guides(fill = guide_legend(override.aes = list(size = 30) ) ) +
  
  
  scale_x_discrete(limits=dfb$Comparison) +
  coord_cartesian(ylim = c(0, 6500)) + 
  scale_y_continuous(expand = c(0,2)) +
  xlab("Comparison") + ylab("Number of genes") + labs(fill='Genes')
  
dev.off()
```

### Up- and downreg. barlots (not used)

```{r fig.height=5, fig.width=7, eval=F}
ggplot() + geom_bar(data=dfb, mapping=aes(x = Comparison, y = DE_genes, fill = group), stat = 'identity',
                    position = position_stacknudge(x=-0.22), width = 0.4) +
  geom_text(data=dfb, mapping=aes(label = DE_genes,
                                       y = DE_genes,
                                       x = Comparison),
            # fontface='bold',
            position = position_stacknudge(x=-0.22, vjust = 0.5, reverse = F), size = 3.5) +
  scale_fill_brewer(palette = 'Set1') +
  new_scale_fill() +
  geom_bar(data=dfm, mapping=aes(x = Comparison, y = DMR_count, fill = group), stat = 'identity',
  position = position_stacknudge(x=0.22), width = 0.4) +
  geom_text(data=dfm, mapping=aes(label = DMR_count,
                                       y = DMR_count,
                                       x = Comparison),
            # fontface='bold',
            position = position_stacknudge(x=0.22, vjust = 0.5), size = 3.5) +
  scale_fill_manual(values = c('darkorange', 'darkolivegreen3', 'red')) +
  theme_classic() + theme(axis.text.x = element_text(angle = 60, vjust = 0.5, size = 16),
                        legend.text = element_text(size = 18), axis.title = element_text(size = 20),
                        legend.title = element_text(size = 20), title = element_text(size = 20)) +
  xlab("Comparison") + ylab("Number of genes or tiles") +
  labs(title = 'Differential expression and methylation of flax genes')


dfb$demr = 'DE'
dfm$demr = 'DMR'

demr = bind_rows(dfb, dfm)
```
